pypsa/linopt.py

```diff
@@ -601,6 +601,165 @@ def set_int_index(ser):
     ser.index = ser.index.str[1:].astype(int)
     return ser
 
+
+def run_and_read_highs(n, problem_fn, solution_fn, solver_logfile,
+                        solver_options, warmstart=None, store_basis=True):
+    """
+    Highs solver function. Reads a linear problem file and passes it to the highs
+    solver. If the solution is feasible the function returns the objective,
+    solution and dual constraint variables. Highs must be installed for usage.
+
+    Installation
+    -------------
+    Installation manual: https://www.maths.ed.ac.uk/hall/HiGHS/
+    After the "make" run, the 'highs' executables are in highs/build/bin/run/highs
+    Once the "ctest" runs make sure you set the path. Path setting notes:
+    >>> highs lib folder must be in "LD_LIBRARY_PATH" environment variable
+    >>> To do this, insert the below path in the .bashrc (hidden in the home folder in Linux/Ubuntu)
+    >>> LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/<your_path>/HiGHS/build/lib
```

Comment:
also the HiGHS executables need to be added to the path. Here are the full installation steps I needed.

```bash
sudo apt-get install cmake # if not installed
git clone git@github.com:ERGO-Code/HiGHS.git
cd HiGHS
mkdir build
cd build
cmake ..
make
ctest
```

Then in `.bashrc` add paths of executables and library:

```sh
export PATH="${PATH}:/foo/HiGHS/build/bin"
export LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:/foo/HiGHS/build/lib"
```

```sh
source .bashrc
```

Now you should see when running `highs`:

```
Running HiGHS 1.1.1 [date: 2021-11-11, git hash: 95342daa]
```